components/compositing/compositor.rs

```diff
@@ -1172,6 +1249,32 @@ impl<Window: WindowMethods> IOCompositor<Window> {
             MouseWindowEvent::MouseDown(_, p) => p,
             MouseWindowEvent::MouseUp(_, p) => p,
         };
+
+        if let Some(ref webrender_api) = self.webrender_api {
+            if let Some(root_pipeline_id) = self.get_root_pipeline_id() {
+                if let Some(root_pipeline) = self.pipeline(root_pipeline_id) {
```

Comment:
Maybe something like:

    let root_pipeline_id = match self.get_root_pipeline_id() {
        Some(root_pipeline_id) => root_pipeline_id,
        None => return,
    };
    let root_pipeline = match self.pipeline(root_pipeline_id) {
        Some(root_pipeline) => root_pipeline,
        None => return,
    };

Just to reduce rightward drift.